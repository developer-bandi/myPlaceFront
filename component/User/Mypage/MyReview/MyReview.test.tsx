import { act, renderHook } from "@testing-library/react";
import axios from "axios";
import { useRouter } from "next/router";
import useMyReview from "./MyReviewHook";

jest.mock("next/router", () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));
const routerMock = {
  pathname: "",
  push: jest.fn(),
};
(useRouter as jest.Mock).mockReturnValue(routerMock);
let status = {
  delete: false,
};
jest.mock("axios", () => {
  return {
    create: jest.fn().mockReturnValue({
      interceptors: {
        request: { use: jest.fn(), eject: jest.fn() },
        response: { use: jest.fn(), eject: jest.fn() },
      },
      delete: jest.fn(() => {
        if (status.delete) {
          return Promise.reject({
            status: 500,
            data: "error",
          });
        }
        return Promise.resolve({
          status: 200,
          data: "data",
        });
      }),
    }),
  };
});

const reviewListMock = {
  count: 1,
  rows: [
    {
      id: "1",
      content: "testContent",
      StoreName: "testStoreName",
      Hashtags: [[1, "test"]] as [number, string][],
      photo: ["test"] as [string],
      createdAt: "testCreatedAt",
    },
  ],
};

describe("MyReview Hook 테스트", () => {
  describe("deleteReview 함수 테스트", () => {
    window.confirm = jest.fn(() => {
      return true;
    });

    it("정상적으로 삭제된경우", async () => {
      window.alert = jest.fn();
      const setServerDataMock = jest.fn();
      const { result } = renderHook(() =>
        useMyReview({
          serverData: { content: reviewListMock, loading: false, error: false },
          setServerData: setServerDataMock,
        })
      );
      await act(async () => {
        await result.current.deleteReview("1");
      });
      expect(window.alert).toBeCalledWith("삭제되었습니다");
      expect(setServerDataMock).toBeCalledWith({
        content: {
          count: 0,
          rows: [],
        },
        loading: false,
        error: false,
      });
    });

    it("에러가 발생한 경우", async () => {
      status.delete = true;
      window.alert = jest.fn();
      const { result } = renderHook(() =>
        useMyReview({
          serverData: { content: reviewListMock, loading: false, error: false },
          setServerData: jest.fn(),
        })
      );
      await act(async () => {
        await result.current.deleteReview("1");
      });
      expect(window.alert).toBeCalledWith("에러가 발생하였습니다");
    });
  });

  it("moveReviewUpdatePage 함수 테스트", () => {
    const { result } = renderHook(() =>
      useMyReview({
        serverData: { content: reviewListMock, loading: false, error: false },
        setServerData: jest.fn(),
      })
    );
    act(() => {
      result.current.moveReviewUpdatePage("1");
    });
    expect(routerMock.push).toBeCalledWith("/contribute/updatereview/1");
  });
});
