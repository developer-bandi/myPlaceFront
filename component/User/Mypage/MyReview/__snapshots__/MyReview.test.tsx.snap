// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyReview Presentational 테스트 노트북사이즈 이하일 경우 결과 없음 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="noResult"
  >
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        작성 후기
      </h1>
      <div
        class="noResult"
      >
        작성한 후기 없음
      </div>
    </div>
  </div>
</div>
`;

exports[`MyReview Presentational 테스트 노트북사이즈 이하일 경우 로딩중 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="loading"
  >
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        작성 후기
      </h1>
      <div
        class="loading"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="loading"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`MyReview Presentational 테스트 노트북사이즈 이하일 경우 에러발생 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="error"
  >
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        작성 후기
      </h1>
      <div
        class="error"
      >
        에러발생
      </div>
    </div>
  </div>
</div>
`;

exports[`MyReview Presentational 테스트 노트북사이즈 이하일 경우 정상 출력 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="result"
  >
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        작성 후기
      </h1>
      <div
        class="reviewBlock"
      >
        <div
          class="titleBlock"
        >
          <div
            class="titleLeftBlock"
          >
            <div
              class="title"
            >
              testStoreName
            </div>
            <div
              class="date"
            >
              testCreatedAt
            </div>
          </div>
          <div
            class="titleRightBlock"
          >
            <div
              class="button"
            >
              삭제
            </div>
            <div
              class="button"
            >
              수정
            </div>
          </div>
        </div>
        <div
          class="content"
        >
          testContent
        </div>
        <div
          class="hashtagList"
        >
          <div
            class="hashtag"
          >
            #
            e
          </div>
          <div
            class="hashtag"
          >
            #
            e
          </div>
        </div>
        <div
          class="photoList"
        >
          <div
            class="photoBlock"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27100%27%20height=%27100%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
      <ul
        class="pageButtonList margin"
      >
        <li>
          <svg
            class="pageIcon ban"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              fill="none"
              points="9 6 15 12 9 18"
              stroke="#000"
              stroke-width="2"
              transform="matrix(-1 0 0 1 24 0)"
            />
          </svg>
        </li>
        <li
          class="pageButton selected"
        >
          1
        </li>
        <li>
          <svg
            class="pageIcon ban"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              fill="none"
              points="9 6 15 12 9 18"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`MyReview Presentational 테스트 데스크톱 일 경우 결과 없음 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="noResult"
  >
    <div
      class="mainBlock"
    >
      <div
        class="subTitle"
      >
        장소찾기
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/bookmark"
          >
            북마크
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/review"
          >
            작성 후기
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        커뮤니티
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/post"
          >
            작성 글
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/comment"
          >
            작성 댓글
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        계정
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/infosetting"
          >
            내 정보
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/withdrawal"
          >
            회원 탈퇴
          </a>
        </li>
      </ul>
    </div>
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        작성 후기
      </h1>
      <div
        class="noResult"
      >
        작성한 후기 없음
      </div>
    </div>
  </div>
</div>
`;

exports[`MyReview Presentational 테스트 데스크톱 일 경우 로딩중 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="loading"
  >
    <div
      class="mainBlock"
    >
      <div
        class="subTitle"
      >
        장소찾기
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/bookmark"
          >
            북마크
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/review"
          >
            작성 후기
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        커뮤니티
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/post"
          >
            작성 글
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/comment"
          >
            작성 댓글
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        계정
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/infosetting"
          >
            내 정보
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/withdrawal"
          >
            회원 탈퇴
          </a>
        </li>
      </ul>
    </div>
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        작성 후기
      </h1>
      <div
        class="loading"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="loading"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`MyReview Presentational 테스트 데스크톱 일 경우 에러발생 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="error"
  >
    <div
      class="mainBlock"
    >
      <div
        class="subTitle"
      >
        장소찾기
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/bookmark"
          >
            북마크
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/review"
          >
            작성 후기
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        커뮤니티
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/post"
          >
            작성 글
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/comment"
          >
            작성 댓글
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        계정
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/infosetting"
          >
            내 정보
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/withdrawal"
          >
            회원 탈퇴
          </a>
        </li>
      </ul>
    </div>
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        작성 후기
      </h1>
      <div
        class="error"
      >
        에러발생
      </div>
    </div>
  </div>
</div>
`;

exports[`MyReview Presentational 테스트 데스크톱 일 경우 정상 출력 1`] = `
<div>
  <div
    class="mainBlock"
    data-testid="result"
  >
    <div
      class="mainBlock"
    >
      <div
        class="subTitle"
      >
        장소찾기
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/bookmark"
          >
            북마크
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/review"
          >
            작성 후기
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        커뮤니티
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/post"
          >
            작성 글
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/comment"
          >
            작성 댓글
          </a>
        </li>
      </ul>
      <div
        class="subTitle"
      >
        계정
      </div>
      <ul
        class="navList"
      >
        <li
          class="nav"
        >
          <a
            href="/user/mypage/infosetting"
          >
            내 정보
          </a>
        </li>
        <li
          class="nav"
        >
          <a
            href="/user/mypage/withdrawal"
          >
            회원 탈퇴
          </a>
        </li>
      </ul>
    </div>
    <div
      class="subBlock"
    >
      <h1
        class="title"
      >
        작성 후기
      </h1>
      <div
        class="reviewBlock"
      >
        <div
          class="titleBlock"
        >
          <div
            class="titleLeftBlock"
          >
            <div
              class="title"
            >
              testStoreName
            </div>
            <div
              class="date"
            >
              testCreatedAt
            </div>
          </div>
          <div
            class="titleRightBlock"
          >
            <div
              class="button"
            >
              삭제
            </div>
            <div
              class="button"
            >
              수정
            </div>
          </div>
        </div>
        <div
          class="content"
        >
          testContent
        </div>
        <div
          class="hashtagList"
        >
          <div
            class="hashtag"
          >
            #
            e
          </div>
          <div
            class="hashtag"
          >
            #
            e
          </div>
        </div>
        <div
          class="photoList"
        >
          <div
            class="photoBlock"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27100%27%20height=%27100%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
      <ul
        class="pageButtonList margin"
      >
        <li>
          <svg
            class="pageIcon ban"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              fill="none"
              points="9 6 15 12 9 18"
              stroke="#000"
              stroke-width="2"
              transform="matrix(-1 0 0 1 24 0)"
            />
          </svg>
        </li>
        <li
          class="pageButton selected"
        >
          1
        </li>
        <li>
          <svg
            class="pageIcon ban"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <polyline
              fill="none"
              points="9 6 15 12 9 18"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
